### ISSUE TITLE
DataLoader gets stuck after model initialization

### ISSUE THREAD

**Issue by anDoer** [pytorch/pytorch#2724]
Everytime I set num_workers > 0 the following code will freeze and never print anything. If I set num_workers=0 it will work though

dataloader = DataLoader(transformed_dataset, batch_size=2, shuffle=True, num_workers=2)
model.cuda()
for i, batch in enumerate(dataloader):
 print(i)

If I put model.cuda() behind the loop, everything will run fine like

dataloader = DataLoader(transformed_dataset, batch_size=2, shuffle=True, num_workers=2)

for i, batch in enumerate(dataloader):
 print(i)
model.cuda()

Does anyone have a solution for that problem?
